src/server.js

```diff
@@ -351,25 +351,21 @@ function initializeClient(socket, client, generateToken, token) {
 		});
 	};
 
-	if (generateToken) {
```

Comment:
This `if` should be kept and replaced with `if (token === null)`. We only want to generate a token if there isn't one.

Reply:
Done, better?

Regarding your `data.token || null` then this `if (token === null)`, we could just rely on `undefined` and not bother about that `|| null`, and do `if (!token)`, etc. Did you do that just to be more explicit? (which is fine by me of course, just checking)

Reply:
Yes explicit null is better.